### PR TITLE
feat(wash): Adds a wit publish command

### PR DESCRIPTION
**Pull request by thomastaylor312**
This is almost exactly the same as `wkg publish` but exposed for convenience within wash

### CODE REVIEW

File: crates/wash-cli/src/wit.rs [hunk 1]
```diff
@@ -66,6 +71,19 @@ pub struct FetchArgs {
     pub common: CommonPackageArgs,
 }
 
+#[derive(Args, Debug, Clone)]
+pub struct PublishArgs {
+    /// The file to publish
+    file: PathBuf,
+
+    /// The registry domain to use. Overrides configuration file(s).
+    #[arg(long = "registry", value_name = "REGISTRY", env = "WKG_REGISTRY")]
```
Comment (joonas) [inline crates/wash-cli/src/wit.rs:80]: Should this be `WASH_REGISTRY`?

Feels a little strange to have `WKG_` variables in `wash` imo.

File: crates/wash-cli/src/wit.rs [hunk 2]
```diff
@@ -110,3 +128,21 @@ impl FetchArgs {
...
+            .client()?
+            .publish_release_file(
+                &self.file,
+                PublishOpts {
+                    registry: self.registry,
+                    ..Default::default()
+                },
+            )
+            .await?;
+        Ok(format!("Published {}@{}", package, version).into())
```
Comment (vados-cosmonic) [inline crates/wash-cli/src/wit.rs:146]:
```suggestion
        Ok(format!("Published {}@{}", package, version).into())
```

Can we have a proper JSON payload for this `CommandOutput` ? Would help with testing it

File: crates/wash-cli/src/wit.rs (hunk 1 above)
Comment (lachieh) [reply crates/wash-cli/src/wit.rs:80]: Is there a need to differentiate between the registry for `wash push`? I don't think that has an env var, but maybe `WASH_WKG_REGISTRY`? Does `wkg` itself use the `WKG_REGISTRY` variable?

Comment (thomastaylor312) [reply crates/wash-cli/src/wit.rs:80]: This is for parity sake with `wkg` so people don't have to have two different env vars to set. I did something similar with the common package opts

File: crates/wash-cli/src/wit.rs (hunk 2 above)
Comment (thomastaylor312) [reply crates/wash-cli/src/wit.rs:146]: I didn't feel like we needed that for something so simple. exit code 0 == published exit code != 0 is failure. This is purely a wrapper around a function that we are using from an upstream library as well, so most testing is done there (the only thing I might want to test is parsing args, but that felt like overkill for this command)

File: crates/wash-cli/src/wit.rs (hunk 1 above)
Comment (thomastaylor312) [reply crates/wash-cli/src/wit.rs:80]: Oh @lachieh's comment wasn't showing up when I wrote my response. He's correct

Comment (vados-cosmonic) [reply crates/wash-cli/src/wit.rs:80]: My vote is for `--wit-registry` & `WASH_WIT_REGISTRY`

**Review by vados-cosmonic — APPROVED** [review]
LGTM :rocket: 

We've talked about the changes here and `WASH_WIT_REGISTRY` along with including the JSON should be great

**Pull request merged**